### PR TITLE
portage: Add support for --quiet-build and --quiet-fail

### PR DESCRIPTION
**Pull request by wltjr**
##### SUMMARY
Added support for options --quiet-build and --quiet-fail

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
portage

##### ANSIBLE VERSION
```
ansible 2.6.0
  config file = None
  configured module search path = ['/home/wlt/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib64/python3.5/site-packages/ansible
  executable location = /usr/lib/python-exec/python3.5/ansible
  python version = 3.5.4 (default, Jan  4 2018, 03:19:06) [GCC 7.2.0]
```

##### ADDITIONAL INFORMATION
May set these by default to Yes/True, along with quiet and set verbose to No/False. I am not sure there is much use for having portage generate output as normal. Seems the default should be quiet across the board, and let the user override as they prefer. It creates a fair amount of noise. I may look to change this in future revisions.

### CODE REVIEW

**Comment by wltjr** [comment]
Reviewers, please double check my modifications to ```mutually_exclusive```. Which technically you can use ```--quiet``` with ```--verbose```. Though may not make sense under Ansible. I am not sure if the others make much sense with verbose. Why I added them, since they should likely not be used together. Only section I am a bit unsure of. Thanks!

File: lib/ansible/modules/packaging/os/portage.py
Comment (webknjaz) [inline lib/ansible/modules/packaging/os/portage.py:182]: I'm afraid, this won't make it in 2.6, because it's a bugfix release and  the next one accepting features should be 2.7.

File: lib/ansible/modules/packaging/os/portage.py
Comment (webknjaz) [inline lib/ansible/modules/packaging/os/portage.py]: Could you please add trailing comma here?

File: lib/ansible/modules/packaging/os/portage.py
Comment (wltjr) [reply lib/ansible/modules/packaging/os/portage.py:182]: Ok I can increase to 2.7, thanks

File: lib/ansible/modules/packaging/os/portage.py
Comment (wltjr) [reply lib/ansible/modules/packaging/os/portage.py]: Sure I missed that, though I think there was not one before. I guess it needed it.

File: lib/ansible/modules/packaging/os/portage.py
Comment (wltjr) [reply lib/ansible/modules/packaging/os/portage.py:182]: Seems I may need to stick with 2.6 or something since CI is failing on 2.7.

**Comment by wltjr** [comment]
shipit

File: lib/ansible/modules/packaging/os/portage.py
Comment (webknjaz) [reply lib/ansible/modules/packaging/os/portage.py]: It's just a thing, which makes diffs nicer when you add/remove list items. Otherwise the other line where you put comma is always listed as changed in diff, which makes it harder to review :)

File: lib/ansible/modules/packaging/os/portage.py
Comment (webknjaz) [reply lib/ansible/modules/packaging/os/portage.py:182]: It will be allowed to merge this branch when devel will start accepting features for 2.7. It's feature-freeze now. I'm going to guess that might happen once we get 2.6 branch separated.

File: lib/ansible/modules/packaging/os/portage.py
Comment (webknjaz) [inline lib/ansible/modules/packaging/os/portage.py:514]: How about `default=None`? Otherwise it might override `EMERGE_DEFAULT_OPTS` values, which would be weird.

File: lib/ansible/modules/packaging/os/portage.py
Comment (wltjr) [reply lib/ansible/modules/packaging/os/portage.py:182]: Ok so 2.7 is correct, and I should just ignore the CI failure till next iteration.

File: lib/ansible/modules/packaging/os/portage.py
Comment (wltjr) [reply lib/ansible/modules/packaging/os/portage.py:514]: I was just being consistent. There are only 3 options that are set to None, all others False. Once merged I plan to switch some of these defaults to True. I feel like ansible should always default to quiet across the board, unless overridden. 

But you make a good point. It maybe worth dropping all False to None. I think its that way based on [the code](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/portage.py#L324). Not familiar with conditionals, but I guess None is the same as False? If that is the case then its moot. Either way I believe it only adds these if set to True. If set to False they do not show up. Or should not per how I read the code.

Comment (webknjaz) [reply lib/ansible/modules/packaging/os/portage.py:514]: No, `None is not False` (yes, it's valid Python statement bwt :smiley:)

Basically, `None` *value* **is casted to False**. *But* `if var` check is using a built-in pythonic protocol for casting bool value out of expression given.
So it does `bool(val)`, which internally does `var.__bool__()` call. And then it uses whatever this method returns for the `if` clause.

`NoneType` (`type(None)`) is always falsy, `[]` is falsy, `0` is falsy, `{}` is falsy. `1` is truthy, `[0]` is truthy, `{'k': 'v'}` is truly.
The end-user can implement whatever logic they want in their classes for casting their instances into boolean.

That is why it's important to check for `val is None` or `val is not None`, sometimes you'll even want `val is False`. `None` is often used for default arg value for functions/methods, which then do `is None` check to ensure the user didn't override them (cause you cannot put mutable defaults in function definitions).

Comment (wltjr) [reply lib/ansible/modules/packaging/os/portage.py:514]: Ok based on that it seems the conditional to add any flag requires it to be set to True, per the [code](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/portage.py#L324).
```python
    for flag, arg in emerge_flags.items():
        if p[flag]:
             args.append(arg)
```

If I set it to None it would not be handled correctly. Or maybe ignored since its not True. Pretty sure that will only evaluate to true, if set to True.

Comment (webknjaz) [reply lib/ansible/modules/packaging/os/portage.py:514]: Yes, it will append flags with `True`, `1`, `1.0`, `y`, `n` (any string) values. And will skip `0`, `0.0`, `None`, `False`, `[]`, `{}`.

Comment (wltjr) [reply lib/ansible/modules/packaging/os/portage.py:514]: So having None vs False is moot for that stuff. Seems only remaining change is the version. I will correct and break CI

Comment (webknjaz) [reply lib/ansible/modules/packaging/os/portage.py:514]: Yes. We'll probably need to consider redesigning defaults separately. Out of scope now.

Comment (wltjr) [reply lib/ansible/modules/packaging/os/portage.py:514]: The module could likely use a few improvements. I am open to such. Not sure how many are using the module. It does not seem very popular, or its meeting alls needs/expectations no clue. Then again not sure many are interested in Gentoo these days...

**Comment by webknjaz** [comment]
@wltjr turns out "no features" rule only applies to core-supported stuff. It's okay to merge in community-supported module.

**Comment by wltjr** [comment]
@webknjaz ok so I guess going back to 2.6 is ok?

**Review by webknjaz — APPROVED** [review]

**Pull request merged**
